nemo/collections/nlp/data/datasets/sgd_dataset/data_processor.py

```diff
@@ -341,19 +382,26 @@ def _create_examples_from_turn(
             examples.append(example)
 
             if service not in prev_states and int(turn_id_) > 0:
-                for slot_name, values in state_update.items():
-                    for value in values:
-                        slot_carryover_values[value].append((service, slot_name))
-                for prev_service, prev_slot_value_list in prev_states.items():
-                    if prev_service == service:
-                        continue
-                    if prev_service in state:
-                        prev_slot_value_list = state[prev_service]
-                    for prev_slot_name, prev_values in prev_slot_value_list.items():
-                        for prev_value in prev_values:
-                            slot_carryover_values[prev_value].append((prev_service, prev_slot_name))
-
-        return examples, states, slot_carryover_values
+                prev_service = ""
+                for prev_s, prev_slot_value_list in prev_states.items():
+                    if prev_s != service:
+                        prev_service = prev_s
+                services_switch_counts[(prev_service, service)] += 1
+
+                if prev_service in states:
+                    prev_slot_value_list = states[prev_service]
+                else:
+                    prev_slot_value_list = prev_states[prev_service]
+
+                cur_slot_value_list = state_update
+                for cur_slot, cur_values in cur_slot_value_list.items():
+                    for prev_slot, prev_values in prev_slot_value_list.items():
+                        if "True" in cur_values or "False" in cur_values:
+                            continue
```

Comment:
what are other possible values in the cur_values besides "True"/"False" ?

Reply:
That one is to ignore the boolean values in counting as they are shared among many slots and it created wrong relations. Values can be anything from categorical slots.